File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/DTOS/UpdateVariantDTO.java
package app.aurafitbackend.DTOS;

import app.aurafitbackend.Beans.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class UpdateVariantDTO {

    private Long id;
    private Size size;
    private Color color;
    private Material material;
    private BigDecimal basePrice;
    private Integer stockQuantity;
    private Product product;
    private List<ProductVariantImage> imagesUrl;
}

File: aurafit-frontend/src/Models/DTOS/UpdateVariantDTO.ts
import {Size} from "../Size.ts";
import {Color} from "../Color.ts";
import {Product} from "../Product.ts";


export class UpdateVariantDTO {
    id: number;
    size: Size;
    color: Color;
    basePrice: number;
    stockQuantity: number;
    product: Product;
    // imagesUrl: string[];

    constructor(id: number, size: Size, color: Color, basePrice: number, stockQuantity: number, product: Product) {
        this.id = id;
        this.size = size;
        this.color = color;
        this.basePrice = basePrice;
        this.stockQuantity = stockQuantity;
        this.product = product;
        // this.imagesUrl = imagesUrl;
    }

}